sentry-ruby/lib/sentry/vernier/profiler.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require "securerandom"
+require_relative "../profiler/helpers"
+require_relative "output"
+
+begin
+  require "vernier"
+rescue LoadError
+end
+
+module Sentry
+  module Vernier
+    class Profiler
+      EMPTY_RESULT = {}.freeze
+
+      attr_reader :started, :event_id, :result
+
+      def initialize(configuration)
+        @event_id = SecureRandom.uuid.delete("-")
+
+        @started = false
+        @sampled = nil
+
+        @profiling_enabled = defined?(Vernier) && configuration.profiling_enabled?
```

Comment:
Is it possible for this part being loaded but `Vernier` is not being defined?

Reply:
@st0012 yes, we rescue from LoadError if vernier is not bundled. It's the same with stackprof, its Profiler gets required even if stackprof is not available (see transaction.rb). I think we can make this whole conditional code loading better in the future and limit the amount of files that sentry requires. ie we can have some API for configuring features and then based on its configuration a smart code loader would know what to do.

Reply:
I think we can `return` in the rescue block then we won't need to call `defined?` at all. For example:

```rb
begin
  require "foo"
rescue LoadError
  return
end

Foo.bar
```

Reply:
sorry to come back to this, but can we keep both similar. Right now stackprof has one way and vernier has another.